dataflow/build.gradle

```diff
@@ -9,23 +9,43 @@ dependencies {
     // ../build.gradle in the shadowJar block).
 }
 
-shadowJar {
-    archiveFileName = "dataflow-shaded.jar"
-    // Without this line, the Maven artifact will have the classifier "all".
-    archiveClassifier = ''
-
-    relocate ('org.checkerframework', 'org.checkerframework.shaded') {
-        // Shade all Checker Framework packages, except for the dataflow qualifiers.
-        exclude 'org.checkerframework.dataflow.qual.*'
-    }
+// Shadowing Test Sources and Dependencies
+import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
+
+/**
+ * Creates a task with name datflow${shadedPkgName} that creates a shaded dataflow jar. The packages will be shaded to
+ * "org.checkerframework.{@code shadedPkgName}" and the jar name is datflow-${shadedPkgName}.jar.
+ * @param shadedPkgName
+ * @return
+ */
+def createDataflowShaded(shadedPkgName) {
+    tasks.create(name: "datflow${shadedPkgName}Jar", type: ShadowJar, dependsOn: compileJava, group: 'Build') {
+        description "Builds datflow-${shadedPkgName}.jar."
+        includeEmptyDirs = false
+        archivesBaseName = "datflow-${shadedPkgName}"
+        // Without this line, the Maven artifact will have the classifier "all".
+        archiveClassifier.set('')
+
+        from shadowJar.source
+        configurations = [project.configurations.runtime]
+
+        destinationDirectory = file("${buildDir}/shadow/datflow${shadedPkgName}")
+
+
+        relocate('org.checkerframework', "org.checkerframework.${shadedPkgName}") {
+            // Shade all Checker Framework packages, except for the dataflow qualifiers.
+            exclude 'org.checkerframework.dataflow.qual.*'
+        }
 
-    // Relocate external dependencies
-    relocate 'org.plume', "org.checkerframework.shaded.org.plume"
+        // Relocate external dependencies
+        relocate 'org.plume', "org.checkerframework.${shadedPkgName}.org.plume"
+    }
 }
 
-artifacts {
-    archives shadowJar
-}
+// Creates a new shaded datflow artifact.  To add a new one, add a new method call below, and add
+// the new jar to publishing and signing blocks.
+createDataflowShaded('shaded')
+createDataflowShaded('NullAway')
```

Comment:
While the official name of the tool is NullAway, in retrospect I wish we had named the github repo just uber/nullaway instead of retaining the capitalization there.  Here, for package naming, I think `nullaway` would be better than `NullAway`